crates/driver/src/tests/setup/blockchain.rs

```diff
@@ -688,55 +684,55 @@ async fn primary_account(web3: &DynWeb3) -> ethcontract::Account {
     ethcontract::Account::Local(web3.eth().accounts().await.unwrap()[0], None)
 }
 
-/// An instance of geth managed by the `dev-geth` crate. When this type is
-/// dropped, the geth instance gets shut down.
-#[derive(Debug)]
-pub struct Geth {
-    port: String,
+/// A blockchain node for development purposes. Dropping this type will
+/// terminate the node.
+pub struct Node {
+    #[allow(dead_code)] // store `process` to later run its `Drop` handler
+    process: tokio::process::Child,
+    url: String,
 }
 
-impl Geth {
-    /// Setup a new geth instance.
-    async fn new() -> Self {
-        let http = reqwest::Client::new();
-        let res = http
-            .post(format!("http://localhost:{DEV_GETH_PORT}"))
-            .send()
-            .await
-            .unwrap();
-        let port = res.text().await.unwrap();
-        Self { port }
-    }
-
-    pub fn url(&self) -> String {
-        format!("http://localhost:{}", self.port)
+impl std::fmt::Debug for Node {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Node").field("url", &self.url).finish()
     }
 }
 
-// What we really want here is "AsyncDrop", which is an unsolved problem in the
-// async ecosystem. As a workaround we create a new runtime so that we can block
-// on the delete request. Spawning a task for this isn't enough because tokio
-// runtimes when they exit drop background tasks, like when a #[tokio::test]
-// function returns.
-impl Drop for Geth {
-    fn drop(&mut self) {
-        let port = std::mem::take(&mut self.port);
-        let task = async move {
-            let client = reqwest::Client::new();
-            client
-                .delete(&format!("http://localhost:{DEV_GETH_PORT}/{port}"))
-                .send()
-                .await
-                .unwrap();
+impl Node {
+    /// Spawn a new node instance.
+    async fn new() -> Self {
+        use tokio::io::AsyncBufReadExt as _;
+
+        let mut process = tokio::process::Command::new("anvil")
```

Comment:
Small suggestion (feel free to ignore):

Would be could to have a `env::var("ANVIL_COMMAND").unwrap_or("anvil")` here or something to allow test runners to control the `anvil` process that runs.

Reply:
This would allow me to have an `anvil.sh` which spawns anvil in `docker`.

Reply:
[67d46ec](https://github.com/cowprotocol/services/pull/1830/commits/67d46ec7c963db331ae443471b89a898917946f1)